coremltools/test/pipeline/test_pipeline.py

```diff
@@ -210,3 +218,52 @@ def test_conversion_bad_inputs(self):
         with self.assertRaises(TypeError):
             model = OneHotEncoder()
             spec = converter.convert(model, "data", "out", "regressor")
+
+
+class TestMakePipeline:
+    @staticmethod
+    def _make_model(input_name, input_length,
+                    output_name, output_length,
+                    convert_to):
+
+        weight_tensor = np.arange(input_length * output_length, dtype='float32')
+        weight_tensor = weight_tensor.reshape(output_length, input_length)
+
+        prog = Program()
+        func_inputs = {input_name: mb.placeholder(shape=(input_length,))}
+        with Function(func_inputs) as ssa_fun:
+            input = ssa_fun.inputs[input_name]
+            y = mb.linear(x=input, weight=weight_tensor, name=output_name)
+            ssa_fun.set_outputs([y])
+            prog.add_function("main", ssa_fun)
+
+        return ct.convert(prog, convert_to=convert_to)
+
+
+    @staticmethod
+    @pytest.mark.parametrize(
+        "model1_backend, model2_backend",
+        itertools.product(["mlprogram", "neuralnetwork"], ["mlprogram", "neuralnetwork"]),
+    )
+    def test_simple(model1_backend, model2_backend):
+        # Create models
+        m1 = TestMakePipeline._make_model("x", 20, "y1", 10, model1_backend)
+        m2 = TestMakePipeline._make_model("y1", 10, "y2", 2, model2_backend)
+
+        # Get non-pipeline result
+        x = np.random.rand(20)
+        y1 = m1.predict({"x": x})["y1"]
+        y2 = m2.predict({"y1": y1})
+
+        pipeline_model = ct.utils.make_pipeline(m1, m2)
+
+        y_pipeline = pipeline_model.predict({"x": x})
```

Comment:
Is there an error handling to make sure that `m1`'s outputs matches `m2`'s inputs?

Reply:
Sadly, this is not possible. Currently, models do not output shapes. So there is no way to check for a mismatch.

Reply:
😨 Maybe have a warning in the doc string? And a TODO to have the check after the output shapes become available?

Reply:
The Core ML Framework enforces that the input and output shapes match (that's why we have to add the output shapes here).

Even when/if output shapes are getting populated, we probably don't want to remove this logic, since we'll want this utility to be able to pipeline models that were created before the output shapes got populated.

Here is what I think would be best - let's only map the input shape to the output shape, if no output shape is present. Thoughts?

Reply:
I've added the fix not to map shapes if it's already present in the output of a model.

Reply:
Sounds great. I have read the logic in `utils.py`.